khal/khalendar/event.py

```diff
@@ -343,6 +351,27 @@ def organizer(self) -> str:
         else:
             return email
 
+    @property
+    def busystatus(self) -> str:
+        if 'X-MICROSOFT-CDO-BUSYSTATUS' not in self._vevents[self.ref]:
+            return 'BUSY'
+        return self._vevents[self.ref]['X-MICROSOFT-CDO-BUSYSTATUS']
+
+    @property
+    def busystatus_formatted(self) -> str:
+        if 'X-MICROSOFT-CDO-BUSYSTATUS' not in self._vevents[self.ref]:
+            return ''
+        busystatus = self._vevents[self.ref]['X-MICROSOFT-CDO-BUSYSTATUS']
+        if busystatus not in self.allowed_busystatus.keys():
+            return ''
+        return self.allowed_busystatus[busystatus]
```

Comment:
**Non blocking hint**: You can use `self._vevents[self.ref].get('X-MICROSOFT-CDO-BUSYSTATUS', None)`, which returns the value (if present) or `None` (if absent).

```suggestion
        busystatus = self._vevents[self.ref].get('X-MICROSOFT-CDO-BUSYSTATUS', None)
        if busystatus not in self.allowed_busystatus.keys():
            return ''
        return self.allowed_busystatus[busystatus]
```

Which can be further simplified into:


```suggestion
        busystatus = self._vevents[self.ref].get('X-MICROSOFT-CDO-BUSYSTATUS', None)
        return self.allowed_busystatus.get(busystatus, ""):
```